core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesCommitManager.java

```diff
@@ -79,11 +83,34 @@ public void commitFileGroups(Set<RewriteFileGroup> fileGroups) {
     }
 
     RewriteFiles rewrite = table.newRewrite().validateFromSnapshot(startingSnapshotId);
-    if (useStartingSequenceNumber) {
-      long sequenceNumber = table.snapshot(startingSnapshotId).sequenceNumber();
-      rewrite.rewriteFiles(rewrittenDataFiles, addedDataFiles, sequenceNumber);
+
+    if (addedDataFiles.size() > 0) {
```

Comment:
I'm not sure why we are making the changes here, shouldn't we just be removing the precondition check
https://github.com/apache/iceberg/blob/f5a753791f4dc6aca78569a14f731feda9edf462/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/FileRewriteCoordinator.java#L58 ?

Reply:
yeah, I was gonna ask the same question. The original logic was implemented before v2 deletes, so I think it make sense to now to just remove that restriction.

Reply:
My initial idea was to rewrite these data files into an empty file or delete the restrictions here, but I'm not sure whether it affects other logic. If it can be removed, I think this is the easiest.